### PR TITLE
Feat/s3 picture of reward

### PR DESCRIPTION
**Pull request by piowit**
https://youtu.be/nO8NTW8LkO4

https://employeerecognitionplatfrom.herokuapp.com/rewards
[test1@test.com](mailto:test1@test.com) / password
[admin@admin.com](mailto:admin@admin.com) / password

check logic, frond-end will be modified to look better later in this pull request

### CODE REVIEW

File: app/controllers/admin/rewards_controller.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@
 module Admin
   class RewardsController < AdminController
     def index
-      @rewards = Reward.all
+      @rewards = Reward.all.order(:title)
```
Comment (ciejj) [inline app/controllers/admin/rewards_controller.rb]: I think there is an N+1 query with the attachments - please check it out
```suggestion
      @rewards = Reward.all.with_attached_photo.order(:title)
```

File: app/searches/reward_search.rb [hunk 2]
```diff
@@ -2,7 +2,8 @@
 
 class RewardSearch < Searchlight::Search
   def base_query
-    Reward.all
+    Reward.all.order(:title).includes([photo_attachment: :blob])
```
Comment (ciejj) [inline app/searches/reward_search.rb]: maybe:
```suggestion
    Reward.all.order(:title).with_attached_photo
```

File: app/views/admin/rewards/index.html.erb [hunk 3]
```diff
@@ -16,6 +17,7 @@
         <td><%= reward.title %></td>
         <td><%= reward.description %></td>
         <td><%= reward.price %></td>
+        <th><% if reward.photo.attached?%> <%= image_tag reward.photo, height: '50px' %> <% end %></th>
```
Comment (ciejj) [inline app/views/admin/rewards/index.html.erb:20]:
```suggestion
        <th><% if reward.photo.attached? %> <%= image_tag reward.photo, height: '50px' %> <% end %></th>
```

File: app/views/rewards/index.html.erb [hunk 4]
```diff
@@ -21,6 +22,7 @@
       <tr data-reward-id="<%= reward.id %>">
         <td><%= reward.title %></td>
         <td><%= reward.price %></td>
+        <th><% if reward.photo.attached?%> <%= image_tag reward.photo, width: '80px' %> <% end %></th>
```
Comment (ciejj) [inline app/views/rewards/index.html.erb]:
```suggestion
        <th><% if reward.photo.attached? %> <%= image_tag reward.photo, width: '80px' %> <% end %></th>
```

**Review by ciejj — APPROVED** [review]
Looks good! 👍

**Review by oskarlakner — APPROVED** [review]
🙌

File: app/controllers/admin/rewards_controller.rb (hunk 1 above)
Comment (piowit) [reply app/controllers/admin/rewards_controller.rb]: true, i forgot to check rewards in admin panel for n+1

**Pull request merged**